src/test/java/org/apache/commons/text/numbers/ParsedDecimalTest.java

```diff
@@ -0,0 +1,764 @@
+        Assertions.assertEquals(exponent, parsed.getExponent());
+        Assertions.assertEquals(digits.length(), parsed.digitCount);
+        Assertions.assertEquals(exponent, parsed.getScientificExponent() - digits.length() + 1);
+    }
+
+    private static void assertThrowsWithMessage(final Executable fn, final Class<? extends Throwable> type,
+            final String msg) {
+        Throwable exc = Assertions.assertThrows(type, fn);
+        Assertions.assertEquals(msg, exc.getMessage());
+    }
+
+    private static double createRandomDouble(final UniformRandomProvider rng) {
+        final long mask = ((1L << 52) - 1) | 1L << 63;
+        final long bits = rng.nextLong() & mask;
+        final long exp = rng.nextInt(2045) + 1;
+        return Double.longBitsToDouble(bits | (exp << 52));
+    }
+
+    /** Get the raw digits in the given decimal as a string.
+     * @param dec decimal instancE
+     * @return decimal digits as a string
+     */
+    private static String digitString(final ParsedDecimal dec) {
+        final StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < dec.digitCount; ++i) {
+            sb.append(dec.digits[i]);
+        }
+        return sb.toString();
+    }
+
+    private static String plainString(final ParsedDecimal dec, final ParsedDecimal.FormatOptions opts) {
+//        try {
+//            StringBuilder sb = new StringBuilder();
+//            dec.toPlainString(sb, opts);
+//            return sb.toString();
+//        } catch (IOException exc) {
+//            throw new UncheckedIOException(exc);
+//        }
+        return dec.toPlainString(opts);
+    }
+
+    private static String scientificString(final ParsedDecimal dec, final ParsedDecimal.FormatOptions opts) {
+//        try {
+//            StringBuilder sb = new StringBuilder();
+//            dec.toScientificString(sb, opts);
+//            return sb.toString();
+//        } catch (IOException exc) {
+//            throw new UncheckedIOException(exc);
+//        }
+        return dec.toScientificString(opts);
+    }
+
+    private static String engineeringString(final ParsedDecimal dec, final ParsedDecimal.FormatOptions opts) {
+//        try {
+//            StringBuilder sb = new StringBuilder();
+//            dec.toEngineeringString(sb, opts);
+//            return sb.toString();
+//        } catch (IOException exc) {
+//            throw new UncheckedIOException(exc);
+//        }
+        return dec.toEngineeringString(opts);
+    }
+
+    private static final class FormatOptionsImpl implements ParsedDecimal.FormatOptions {
+
+        private boolean includeFractionPlaceholder = true;
+
+        private boolean signedZero = true;
+
+        private char[] digits = "0123456789".toCharArray();
+
+        private char decimalSeparator = '.';
+
+        private char thousandsGroupingSeparator = ',';
+
+        private boolean groupThousands = false;
+
+        private char minusSign = '-';
```

Comment:
I added this since it is part of `DecimalFormatSymbols`. I believe there are locales/languages that use a different symbol for minus but I can't think of any examples right now.